SingularityUI/app/components/common/JSONButton.jsx

```diff
@@ -27,10 +27,9 @@ export default class JSONButton extends Component {
 
     this.showJSON = this.showJSON.bind(this);
     this.hideJSON = this.hideJSON.bind(this);
-  }
 
-  componentDidMount() {
-    this.clipboard = new Clipboard('.copy-btn');
+    this.attachClipboard = this.attachClipboard.bind(this);
```

Comment:
You can bind the functions in the constructor along with the `showJSON` and `hideJSON` functions. There's no reason to do this in componentDidMount. If you have lots of functions that you need to bind, you can use `_.bindAll`